core_java/jdbc/src/main/java/ca/jrvs/apps/jdbc/CustomerDAO.java

```diff
@@ -0,0 +1,184 @@
+      "email = ?, phone = ?, address = ?, city = ?, state = ?, zipcode = ? WHERE customer_id = ?";
+  private static final String DELETE = "DELETE FROM customer WHERE customer_id = ?";
+  private static final String GET_ALL_LMT = "SELECT customer_id, first_name, last_name, email, phone, " +
+      "address, city, state, zipcode FROM customer ORDER BY last_name, first_name LIMIT ?";
+  private static final String GET_ALL_PAGED = "SELECT customer_id, first_name, last_name, email, phone, " +
+      "address, city, state, zipcode FROM customer ORDER BY last_name, first_name LIMIT ? OFFSET ?";
+
+  //Start the connection
+  public CustomerDAO(Connection connection) {
+    super(connection);
+  }
+
+  @Override
+  //find customer by ID.
+  public Customer findById(long id) {
+    Customer customer = new Customer();
+    try(PreparedStatement statement = this.connection.prepareStatement(GET_ONE);) {
+      statement.setLong(1, id);
+      ResultSet rs = statement.executeQuery();
+      while(rs.next()){
+        customer.setId(rs.getLong("customer_id"));
+        customer.setFirstName(rs.getString("first_name"));
+        customer.setLastName(rs.getString("last_name"));
+        customer.setEmail(rs.getString("email"));
+        customer.setPhone(rs.getString("phone"));
+        customer.setAddress(rs.getString("address"));
+        customer.setCity(rs.getString("city"));
+        customer.setState(rs.getString("state"));
+        customer.setZipCode(rs.getString("zipcode"));
+      }
+    }catch (SQLException e){
+      e.printStackTrace();
+      throw new RuntimeException(e);
+    }
+    return customer;
+  }
+
+  @Override
+  public List<Customer> findAll() {
+    return null;
+  }
+
+  @Override
+  //update customer information
+  public Customer update(Customer dto) {
+    Customer customer = null;
+    try{
+      this.connection.setAutoCommit(false);
+    }catch (SQLException e){
+      e.printStackTrace();
+      throw new RuntimeException(e);
+    }
+    try(PreparedStatement statement = this.connection.prepareStatement(UPDATE);){
+      statement.setString(1, dto.getFirstName());
+      statement.setString(2, dto.getLastName());
+      statement.setString(3, dto.getEmail());
+      statement.setString(4, dto.getPhone());
+      statement.setString(5, dto.getAddress());
+      statement.setString(6, dto.getCity());
+      statement.setString(7, dto.getState());
+      statement.setString(8, dto.getZipCode());
+      statement.setLong(9, dto.getId());
+      statement.execute();
+      this.connection.commit();
+      customer = this.findById(dto.getId());
+    }catch(SQLException e){
+      try{
+        this.connection.rollback();
+      }catch (SQLException sqle){
+        e.printStackTrace();
+        throw new RuntimeException(sqle);
+      }
+      e.printStackTrace();
+      throw new RuntimeException(e);
+    }
+    return customer;
+  }
+
+  @Override
+  //create new customer and insert into database
```

Comment:
please use property javadoc to document method
e.g.     /**
     * Constructs a new, empty set; the backing <tt>HashMap</tt> instance has
     * the specified initial capacity and the specified load factor.
     *
     * @param      initialCapacity   the initial capacity of the hash map
     * @param      loadFactor        the load factor of the hash map
     * @throws     IllegalArgumentException if the initial capacity is less
     *             than zero, or if the load factor is nonpositive
     */


You can generate javadoc using IDE (e.g. `/**` and enter)

Reply:
please go through your code and fix similar issues